src/app/components/point/point.component.ts

```diff
@@ -1,38 +1,63 @@
-import { AsyncPipe, NgIf } from '@angular/common';
-import { Component } from '@angular/core';
-import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
-import { TetrisQuery } from '@angular-tetris/state/tetris/tetris.query';
-import { Observable, of, timer } from 'rxjs';
-import { map, switchMap } from 'rxjs/operators';
+import { TetrisStateService } from '@angular-tetris/state/tetris/tetris.state';
+import { NgIf } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  DestroyRef,
+  effect,
+  inject,
+  signal
+} from '@angular/core';
+import { UntilDestroy } from '@ngneat/until-destroy';
 import { NumberComponent } from '../number/number.component';
 
 const REFRESH_LABEL_INTERVAL = 3000;
 @UntilDestroy()
 @Component({
   selector: 't-point',
   standalone: true,
-  imports: [NumberComponent, NgIf, AsyncPipe],
+  imports: [NumberComponent, NgIf],
   templateUrl: './point.component.html',
-  styleUrls: ['./point.component.scss']
+  styleUrls: ['./point.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class PointComponent {
-  labelAndPoints$: Observable<LabelAndNumber> = this.query.hasCurrent$.pipe(
-    untilDestroyed(this),
-    switchMap((hasCurrent) => {
-      if (hasCurrent) {
-        return of(new LabelAndNumber('Score', this.query.raw.points));
-      }
-      return timer(0, REFRESH_LABEL_INTERVAL).pipe(
-        map((val) => {
-          const isOdd = val % 2 === 0;
-          const { points, max } = this.query.raw;
-          return isOdd ? new LabelAndNumber('Score', points) : new LabelAndNumber('Max ', max);
-        })
-      );
-    })
-  );
```

Comment:
Let's stick with the old RxJS implementation here as well, looks so much easy to understand.